src/Database/Bolt/Transaction.hs

```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Database.Bolt.Transaction
+  ( TxError
+  , transact
+  , transact_
+  )
+where
+
+import           Database.Bolt.Connection.Pipe
+import           Database.Bolt.Connection.Type
+import           Database.Bolt.Connection       ( BoltActionT
+                                                , queryP_
+                                                , queryP'
+                                                , run
+                                                )
+import           Database.Bolt.Record
+import           Database.Bolt.Value.Type
+
+import           Control.Exception              ( IOException )
+import           Control.Monad                  ( join )
+import           Control.Monad.Catch            ( Exception
+                                                , catch
+                                                , throwM
+                                                )
+import           Control.Monad.IO.Class         ( MonadIO(..)
+                                                , liftIO
+                                                )
+import           Control.Monad.Trans.Reader     ( ask )
+import           Data.Functor                   ( void )
+import           Data.Map                       ( Map )
+import           Data.Map.Strict                ( empty )
+import           Data.Text                      ( Text )
+
+data TxError = TxError deriving Show
+instance Exception TxError
+
+-- |Runs a list of cypher queries with parameters as an atomic operation and returns all the records
+transact :: MonadIO m => [Cypher] -> BoltActionT m [Record]
+transact = transaction False
```

Comment:
If you use `Cypher` type for `transact(_)`, I think it's better to use it in `query(..)`s too. This would break some backward capability, but would be more consistent.

Reply:
I was going to propose that too :) Will do it then.